test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java

```diff
@@ -92,21 +92,9 @@ public void testFailure(Failure failure) throws Exception {
         GradleMessageBuilder gradleMessageBuilder = new GradleMessageBuilder(b);
         gradleMessageBuilder.appendAllOpts(failure.getDescription());
 
-        if (isRestApiCompatibilityTest()) {
```

Comment:
Why are we removing this? Rest compatibility tests are still relevant to 9.0.

Reply:
It seems `tests.restCompat` property is used in the tests only to select nodes for V7 compatibility tests (https://github.com/elastic/elasticsearch/pull/117992/files#diff-8563be9d0ca3fe7e4456eb5712dd42588425972b9848913734f5e177f9b25fa8L333), and this mode is no longer supported in V9

Reply:
Rest compatibility mode is still a thing in V9, it's just that we only support back to V8, and we still run these tests ([for example](https://gradle-enterprise.elastic.co/s/3ip4zgdx5ua2m)). This infra needs to remain.

Reply:
Sorry, I missed the case where this branch might still be relevant. I've reverted the changes